utoipauto-core/src/file_utils.rs

```diff
@@ -81,19 +81,34 @@ pub fn extract_module_name_from_path(path: &str, crate_name: &str) -> String {
     // `./crates/subcrate/src/my/module`, etc., so we need to remove anything up to `src`
     // (or `tests`) to still produce `crate::my::module`.
     // So we split the segments by the last occurrence of `src` or `tests` and take the last part.
-    let segments_inside_crate = match segments
-        .iter()
-        .rposition(|&segment| segment == "src" || segment == "tests")
-    {
-        Some(idx) => &segments[(idx + 1)..],
-        None => &segments,
+    let segments_inside_crate = find_segment_and_skip(&segments, &["src", "tests"], 1);
+
+    // Also skip fragments that are already out of the crate name. For example,
+    // `./src/lib/my/module/name from crate::my::module` should turn into `crate::my::module:name`,
+    // and not into `crate::lib::my::module::name`.
+    let mut crate_segments = crate_name.split("::");
+    let first_crate_fragment = crate_segments.next().unwrap();
```

Comment:
I am not a huge fan of unwraps.
Please replace it with expect and a descriptive error message

Reply:
Done